components/tools/OmeroPy/test/integration/test_icontainer.py

```diff
@@ -45,116 +42,51 @@ def testFindAnnotations(self):
         i = ipojo.createDataObject(i, None)
 
     def testFindAndCountAnnotationsForSharedData(self):
-        uuid = self.root.sf.getAdminService().getEventContext().sessionUuid
-        admin = self.root.sf.getAdminService()
-
-        # create new users
-        # group1
-        new_gr1 = ExperimenterGroupI()
-        new_gr1.name = rstring("group1_%s" % uuid)
-        new_gr1.ldap = rbool(False)
-        gid = admin.createGroup(new_gr1)
-
-        # new user1
-        new_exp = ExperimenterI()
-        new_exp.omeName = rstring("user1_%s" % uuid)
-        new_exp.firstName = rstring("New")
-        new_exp.lastName = rstring("Test")
-        new_exp.ldap = rbool(False)
-
-        defaultGroup = admin.getGroup(gid)
-        listOfGroups = list()
-        listOfGroups.append(admin.lookupGroup("user"))
-
-        eid = admin.createExperimenterWithPassword(
-            new_exp, rstring("ome"), defaultGroup, listOfGroups)
-
-        # new user2
-        new_exp2 = ExperimenterI()
-        new_exp2.omeName = rstring("user2_%s" % uuid)
-        new_exp2.firstName = rstring("New2")
-        new_exp2.lastName = rstring("Test2")
-        new_exp2.ldap = rbool(False)
-
-        defaultGroup = admin.getGroup(gid)
-        listOfGroups = list()
-        listOfGroups.append(admin.lookupGroup("user"))
-
-        eid2 = admin.createExperimenterWithPassword(
-            new_exp2, rstring("ome"), defaultGroup, listOfGroups)
-
-        # get users
-        user1 = admin.getExperimenter(eid)
-        user2 = admin.getExperimenter(eid2)
-
-        # login as user1
-        cl1 = self.new_client(user=user1, password="ome")
-        update1 = cl1.sf.getUpdateService()
-        ipojo1 = cl1.sf.getContainerService()
 
-        # create image
-        img = ImageI()
-        img.setName(rstring('test1154-img-%s' % (uuid)))
+        # create users
+        group = self.new_group(perms="rwra--")
+        client1, user1 = self.new_client_and_user(group=group)
 
-        # default permission 'rw----':
-        img = update1.saveAndReturnObject(img)
+        # create image
+        img = self.make_image(name='test1154-img-%s' % self.uuid(),
+                              client=client1)
         img.unload()
 
+        update1 = client1.sf.getUpdateService()
         ann1 = CommentAnnotationI()
-        ann1.textValue = rstring("user comment - %s" % uuid)
+        ann1.textValue = rstring("user comment - %s" % self.uuid())
         l_ann1 = ImageAnnotationLinkI()
         l_ann1.setParent(img)
         l_ann1.setChild(ann1)
         update1.saveObject(l_ann1)
```

Comment:
`Itest.link()` only applies to `ProjectDatasetLinkI` and `DatasetImageLinkI` at the moment. As discussed earlier, extending it to cover `ObjectAnnotationLinkI` could be one of the next library helper refactoring PRs since it is widely used across the code.